frb_example/benches/rust/src/executor.rs

```diff
@@ -0,0 +1,237 @@
+        value: Some(elapsed),
+        extra: None,
+        unit,
+    });
+}
+
+/// simple newtype pattern
+///
+/// this is the bridge's custom [error handler](http://cjycode.com/flutter_rust_bridge/feature/handler.html#example-report-errors-to-your-backend-in-addition-to-telling-dart).
+#[derive(Clone, Copy)]
+pub struct BenchErrorHandler(ReportDartErrorHandler);
+
+impl Default for BenchErrorHandler {
+    fn default() -> Self {
+        Self(ReportDartErrorHandler)
+    }
+}
+
+/// trait bound required by [`SimpleHandler`]
+impl ErrorHandler for BenchErrorHandler {
+    fn handle_error(&self, port: MessagePort, error: handler::Error) {
+        self.0.handle_error(port, error)
+    }
+
+    fn handle_error_sync(
+        &self,
+        error: flutter_rust_bridge::handler::Error,
+    ) -> flutter_rust_bridge::support::WireSyncReturnStruct {
+        self.0.handle_error_sync(error)
+    }
+}
+
+/// a slightly more elaborated newtype
+///
+/// this is the bridge's custom [executor](http://cjycode.com/flutter_rust_bridge/feature/handler.html#example-log-when-execution-starts-and-ends).
+pub struct BenchExecutor {
+    inner: ThreadPoolExecutor<BenchErrorHandler>,
+    json: bool,
+}
+
+/// subscribe for tracing only on native platforms
+#[cfg(not(target_family = "wasm"))]
+fn maybe_subscribe(json: bool) {
+    use tracing_subscriber::FmtSubscriber;
+    let subscriber = if json {
+        None
+    } else {
+        Some(
+            FmtSubscriber::builder()
+                .with_max_level(tracing::Level::TRACE)
+                .finish(),
+        )
+    };
+    if let Some(subscriber) = subscriber {
+        tracing::subscriber::set_global_default(subscriber)
+            .expect("Setting default subscriber failed");
+    }
+}
+
+/// on web platform, this is a no-op
+#[cfg(target_family = "wasm")]
+#[inline(always)]
+fn maybe_subscribe(_: bool) {}
+
+impl BenchExecutor {
+    /// also retrieve env var to set mode to **stdout** or **json**
+    /// and subscribe to `tracing` if needed
+    pub(crate) fn new(error_handler: BenchErrorHandler) -> Self {
+        let json = std::env::var("JSON")
+            .unwrap_or_else(|_| "false".into())
+            .parse::<bool>()
+            .expect("Invalid JSON env var (expected boolean)");
+        maybe_subscribe(json);
+        Self {
+            inner: ThreadPoolExecutor::new(error_handler),
+            json,
+        }
+    }
+}
+
+impl Executor for BenchExecutor {
+    /// wrap [`bench_around`] over async FFI calls
+    fn execute<TaskFn, TaskRet>(&self, wrap_info: WrapInfo, task: TaskFn)
+    where
+        TaskFn: FnOnce(TaskCallback) -> anyhow::Result<TaskRet> + Send + UnwindSafe + 'static,
+        TaskRet: IntoDart,
+    {
+        let debug_name_string = wrap_info.debug_name.to_string();
+        let json = self.json;
+        self.inner.execute(wrap_info, move |task_callback| {
+            Self::bench_around(&debug_name_string, json, move || task(task_callback))
```

Comment:
Btw, how `criterion`, a popular rust micro-benchmark lib, benchmarks things: 
https://bheisler.github.io/criterion.rs/book/analysis.html Maybe helpful to get some insights about benchmarking

Reply:
Sadly I don't have time to implement a high-quality benchmarking lib either

Reply:
Definitely not to reinvent bench lib :) See comments above